samples/cpp/benchmark_app/main.cpp

```diff
@@ -299,9 +306,19 @@ int main(int argc, char* argv[]) {
                                   "but it still may be non-optimal for some cases, for more "
                                   "information look at README."
                                << slog::endl;
-                    if (std::string::npos == device.find("MYRIAD"))  // MYRIAD sets the default number of
-                                                                     // streams implicitly (without _AUTO)
-                        device_config[key] = std::string(getDeviceTypeFromName(device) + "_THROUGHPUT_AUTO");
+                    if (std::string::npos == device.find("MYRIAD")) { // MYRIAD sets the default number of
+                                                                      // streams implicitly (without _AUTO)
+                        auto supported_properties = core.get_property(device, ov::supported_properties);
+                        if (std::find(supported_properties.begin(), supported_properties.end(), key) !=
+                            supported_properties.end()) {
+                            device_config[key] = std::string(getDeviceTypeFromName(device) + "_THROUGHPUT_AUTO");
+                        } else if (std::find(supported_properties.begin(), supported_properties.end(), ov::streams::num.name()) !=
+                                   supported_properties.end()) {
+                            // Use API 2.0 key for streams
+                            key = ov::streams::num.name();
+                            device_config[key] = std::to_string(ov::streams::AUTO);
+                        }
+                    }
```

Comment:
Can we combine this code with setThroughputStreams (mybe put come checks for MYRIAD inside). And please extract this code as separate function instead of lambda (if won't require too much efforts)

Reply:
Hm, this code is already a part of setThroughputStreams() lambda. Can we postpone with extracting this lambda to independent function, since it would be better to merge this PR ASAP to fix problems with streams in benchmark_app?

Reply:
OK. But please make this lambda reusable for both cases (MYRIAD and other)